src/Symfony/Component/Console/Application.php

```diff
@@ -974,22 +974,31 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
             }
         }
 
-        if ($command instanceof SignalableCommandInterface && ($this->signalsToDispatchEvent || $command->getSubscribedSignals())) {
-            if (!$this->signalRegistry) {
-                throw new RuntimeException('Unable to subscribe to signal events. Make sure that the `pcntl` extension is installed and that "pcntl_*" functions are not disabled by your php.ini\'s "disable_functions" directive.');
-            }
+        if ($this->signalsToDispatchEvent) {
+            $commandSignals = $command instanceof SignalableCommandInterface ? $command->getSubscribedSignals() : [];
+            $dispatchSignals = $this->dispatcher && $this->dispatcher->hasListeners(ConsoleEvents::SIGNAL);
```

Comment:
$this->dispatcher is typed to be a `\Symfony\Contracts\EventDispatcher\EventDispatcherInterface`, but `hasListeners` is not part of that contract. 

This leads to our package breaking with the current branch version (5.4.x-dev in our case), since it does not fulfill the `\Symfony\Component\EventDispatcher\EventDispatcherInterface`